cloud/services/subnets/subnets.go

```diff
@@ -58,100 +47,100 @@ func (s *Service) getExisting(ctx context.Context, rgName string, spec *Spec) (*
 }
 
 // Reconcile gets/creates/updates a subnet.
-func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
-	subnetSpec, ok := spec.(*Spec)
-	if !ok {
-		return errors.New("Invalid Subnet Specification")
-	}
-	existingSubnet, err := s.getExisting(ctx, s.Scope.Vnet().ResourceGroup, subnetSpec)
-	if err == nil {
-		// subnet already exists, update the spec and skip creation
-		var subnet *infrav1.SubnetSpec
-		if subnetSpec.Role == infrav1.SubnetControlPlane {
-			subnet = s.Scope.ControlPlaneSubnet()
-		} else if subnetSpec.Role == infrav1.SubnetNode {
-			subnet = s.Scope.NodeSubnet()
-		} else {
+func (s *Service) Reconcile(ctx context.Context) error {
+	for i, subnetSpec := range s.Scope.SubnetSpecs() {
+		existingSubnet, err := s.getExisting(ctx, s.Scope.Vnet().ResourceGroup, subnetSpec)
+		if err == nil {
+			// subnet already exists, update the spec and skip creation
+			var subnet *infrav1.SubnetSpec
+			if subnetSpec.Role == infrav1.SubnetControlPlane {
+				subnet = s.Scope.ControlPlaneSubnet()
+			} else if subnetSpec.Role == infrav1.SubnetNode {
+				subnet = s.Scope.NodeSubnet()
+			} else {
+				return nil
+			}
+
+			subnet.Role = subnetSpec.Role
+			subnet.Name = existingSubnet.Name
+			subnet.CidrBlock = existingSubnet.CidrBlock
+			subnet.ID = existingSubnet.ID
+
+			// Continue to iterating the Subnet spec array
+			if i != len(s.Scope.SubnetSpecs())-1 {
```

Comment:
why is this needed?

Reply:
because the reconcile receives an array in the case one item for the control-plane subnet and the other for the node subnet, if we not do that, it will apply one for the first one always.
applying the update for only for the first item in the array is something we want?

Reply:
Got it, good catch. Thanks for adding a unit test as well.